tests/test_apply.py

```diff
@@ -1,16 +1,179 @@
-import tempfile
 import os
-import boto3
-import uuid
+import random
 import subprocess
+import tempfile
+import uuid
 from typing import Dict
 
+import boto3
+
 THIS_PATH = os.path.abspath(os.path.dirname(__file__))
 ROOT_PATH = os.path.join(THIS_PATH, "..")
 TFLOCAL_BIN = os.path.join(ROOT_PATH, "bin", "tflocal")
 LOCALSTACK_ENDPOINT = "http://localhost:4566"
 
 
+def test_access_key_override_fallback(monkeypatch):
+    access_key = f"{mock_access_key()}"
```

Comment:
nit: could be simplified to:
```suggestion
    access_key = mock_access_key()
```